scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,235 @@
+
+        if len(response_json) > 1:
+            raise Exception(
+                "Unable to get a unique change for {}{}."
+                "This can also happen if the same change-id is used "
+                "in multiple gerrit projects.".format(
+                    change_id,
+                    " and branch {}".format(branch) if branch else '')
+            )
+        elif len(response_json) != 1:
+            raise Exception("Unable to find a change for {}{}".format(
+                change_id, " and branch {}".format(branch) if branch else ''))
+
+        self._change_object = response_json[0]
+
+    def _set_change_object(self, change_object):
+        self._change_object = change_object
+
+    def _load_change_object(self, patchset=None):
+        self.id = str(self._change_object['_number'])
+        self.change_id = self._change_object['change_id']
+        self.gerrit_project = self._change_object['project'].split('/')[1]
+        self.status = self._change_object['status']
+        if not patchset:
+            current_revision = self._change_object['current_revision']
+            revision_obj = self._change_object['revisions'][current_revision]
+        else:
+            revisions = [r for r in self._change_object['revisions'].values()
+                         if r['_number'] == int(patchset)]
+            if len(revisions) != 1:
+                print json.dumps(self._change_object, indent=4)
+                raise Exception(
+                    "Unable to find patchset {} for change {} ".format(
+                        patchset, self.id))
+            revision_obj = revisions[0]
+        self.patchset = str(revision_obj['_number'])
+        self.url = revision_obj['fetch']['anonymous http']['url']
+        self.ref = revision_obj['fetch']['anonymous http']['ref']
+        self.branch = self._change_object['branch']
+        self.subject = revision_obj['commit']['subject']
+        self.commit_message = revision_obj['commit']['message']
+        self.mergeable = self._change_object.get('mergeable', False)
+        self.submittable = self._change_object.get('submittable', False)
+        self.labels = self._change_object['labels']
+
+    def load_dependencies(self, depth=None, loaded_deps=[]):
+        if depth == 0:
+            return
+        elif depth is not None:
+            depth -= 1
+        deps = find_dependency_headers(self.commit_message)
+        for change_id in deps:
+            change = GerritChange(change_id, self.branch)
+            if change.id in loaded_deps:
+                continue
+            self._dependencies.append(change)
+            loaded_deps.append(change.id)
+            change.load_dependencies(depth, loaded_deps)
+
+    def dependencies(self):
+        return self._dependencies[:]
+
+    def depends_on(self, change):
+        deps = find_dependency_headers(self.commit_message)
+        return change.id in deps or change.change_id in deps
```

Comment:
This won't work when the depends-on format is a URL. We probably need to compare this to the self._dependencies. We should also consider if we should search grand-child dependencies.

Reply:
I don't think that is correct. `find_dependency_headers` always returns a list of numeric IDs or extended change IDs, even when the depends-on value is an url.
Regarding grand-child dependencies, I updated the method name and docstring to reflect the fact that this only covers direct dependencies.

Reply:
Right, I think I was meaning to make sure we have both formats covered, however we are checking both change.id and change.change_id which I must have missed.

As an aside, we should really disambiguate between the two. Gerrit uses a "change-id" to mean multiple things[0] when referring to an identifier. An identifier can be made up of <project> <branch> and <Change-Id> or <Numberic-Id>. So long as the identifier is unambiguous or unique gerrit does the right thing. In general though it uses the term "change-number" for the numeric representation which is likely something we should use.

We could do this as as a follow up though.

[0] https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#change-id